### PR TITLE
content-type header parsing: be more forgiving

### PR DESCRIPTION
**Pull request by matthewleon**
The space after the semicolon delimiter in HTTP header content is optional according to the standard, and a number of implementations will not add such whitespace.

I ran into this problem when mocking form data using the form-data lib (https://www.npmjs.com/package/form-data), which does not add the optional space when serializing headers.

Please let me know if it's worth adding a test _just_ to demonstrate the tolerance of "no-whitespace" headers. I figured that passing the current test should be enough.

### CODE REVIEW

**Comment by matthewleon** [comment]
@kettanaito this test failure appears to be non-deterministic. At least, I can't replicate locally. Running the integration tests on this branch results in no failures.

File: src/utils/internal/parseMultipartData.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export function parseMultipartData<T extends DefaultRequestMultipartBody>(
     return undefined
   }
 
-  const [, ...directives] = contentType.split('; ')
+  const [, ...directives] = contentType.split(/; */)
```
Comment (kettanaito) [inline src/utils/internal/parseMultipartData.ts:56]: This is a great change! Let's add a test for handling such headers in `src/utils/internal/parseMultipartData.test.ts`. Please, would you be comfortable with adding such test?

**Review by kettanaito — COMMENTED** [review]
Thank you for this improvement, @matthewleon! 
No worries about the failing test, it doesn't seem to be related to your changes. I've restarted the CI so it should pass now. 

I've left just one comment about the need to test this.

**Comment by matthewleon** [comment]
@kettanaito This should be good to go.

File: src/utils/internal/parseMultipartData.test.ts [hunk 2]
```diff
@@ -1,6 +1,24 @@
 import { parseMultipartData } from './parseMultipartData'
 
 test('parses a given valid multipart string', async () => {
+  expect.assertions(3)
+  await testMultipartDataWithContentType(
+    'multipart/form-data; boundary=WebKitFormBoundaryvZ1cVXWyK0ilQdab',
+  )
+})
+
+test('parses a given valid multipart string give non-pretty content-type', async () => {
```
Comment (kettanaito) [inline src/utils/internal/parseMultipartData.test.ts]: Looks like we need one more scenario when there are multiple spaces between the content type and its additional information: `multipart/form-data;___boundary=xxx` (replace "_" by spaces, markdown trims that).

Comment (matthewleon) [reply src/utils/internal/parseMultipartData.test.ts]: done

**Review by kettanaito — APPROVED** [review]
Thank you for shipping this improvement, @matthewleon!

**Comment by kettanaito** [comment]
Welcome to contributors 🎉
